### ISSUE TITLE
Option to remove @ file:///... when building wheel

### ISSUE THREAD

**Issue by Matheus-Ribeiro-ITA** [pdm-project/pdm#1895]
Hi, Frost and PDM team!

I would like to suggest the option to the user to remove the local path when building packages. 

For example, let`s say I have an APP with PDM that uses 2 local libraries: package_A and package_B

would be awesome to have on the app `pyproject.toml`:
```
dependencies = [
    "package_A @ file:///${PROJECT_ROOT}/package_A",
    "package_B @ file:///${PROJECT_ROOT}/package_B",
]
```

but after building the app wheel we could have something like:

```
dependencies = [
    "package_A>=1.2.0",
    "package_B>2.0.0",
]
```

The reason behind it is that I want to be able to test package_A and B on my local development. But on production, I will actually install the wheel directly, e.g on Databricks-jobs clusters, so I actually don`t use the local path because I publish the package_A, B and the App to an artifactory.

So in summary:

- **Local enviroment**: have  relative paths.
- An option `Pdm build --remove_relative_path`:  can remove the relative paths and add the package version.
- **Production environment**: A wheel without a relative path.

Would love to hear the community's opinion and if there is already a similar solution. 

If this doesn`t make sense to the core code, let me know pls, no problem. I will try to do a plugin too for it, but think could be useful on the main PDM too.

**Comment by frostming** [comment]
You can just duplicate the dependencies in both production(normal deps with versions) and dev dependencies(local deps)

**Comment by Matheus-Ribeiro-ITA** [comment]
Awesome, thanks for your quick response!

You are totally right! 

In my case, I don't have access to the artifacts on my local dev env, so pdm build fails because it tries to lock the dependencies before. Which I agree is the correct behavior.

Let me know If anyone has a better idea, otherwise, I will create a custom plugin or script to skip lock, something like `pdm build --skip-lock`.


Thanks for the insights and help
